File: add_animations.mjs
import fs from 'fs';
import path from 'path';

function processFile(filePath) {
    let content = fs.readFileSync(filePath, 'utf8');

    // Skip if already animated
    if (content.includes('containerVariants')) {
        return;
    }
    
    let modified = false;

    // 1. Add import
    if (!content.includes("import { motion }")) {
        // Try to insert after imports
        content = content.replace(/(import .*?;)/, "$1\nimport { motion } from 'framer-motion';");
        modified = true;
    }

    // 2. Add variants before return (
    const returnRegex = /(return\s*\(\s*<AppLayout[\s\S]*?>)/;
    if (returnRegex.test(content)) {
        const variants = `
  const containerVariants = {
    hidden: { opacity: 0 },
    show: {
      opacity: 1,
      transition: { staggerChildren: 0.1, delayChildren: 0.1 },
    },
  };

  const itemVariants = {
    hidden: { opacity: 0, y: 20 },
    show: {
      opacity: 1,
      y: 0,
      transition: { type: 'spring', stiffness: 100, damping: 20 },
    },
  };

  `;
        content = content.replace(returnRegex, variants + '$1');
        modified = true;
    }

    // 3. Replace the main container right after <Head ... />
    // This looks for `<Head... />` followed by some spaces/newlines, then a `<div className="...">`
    // We replace it with `<motion.div variants={containerVariants} initial="hidden" animate="show" className="...">`
    const mainDivRegex = /(<Head[^>]*\/>\s*)<div(\s+className="[^"]+")>/;
    if (mainDivRegex.test(content)) {
        content = content.replace(mainDivRegex, '$1<motion.div variants={containerVariants} initial="hidden" animate="show"$2>');
        
        // Replace the last </div> before </AppLayout> with </motion.div>
        const appLayoutEndRegex = /<\/div>(\s*<\/AppLayout>)/;
        content = content.replace(appLayoutEndRegex, '</motion.div>$1');
        modified = true;
    }

    // 4. Wrap inner elements: <PageHeader, <ContentPanel
    const pageHeaderRegex = /(<PageHeader[\s\S]*?<\/PageHeader>)/g;
    content = content.replace(pageHeaderRegex, '<motion.div variants={itemVariants}>\n$1\n</motion.div>');

    const contentPanelRegex = /(<ContentPanel[\s\S]*?<\/ContentPanel>)/g;
    content = content.replace(contentPanelRegex, '<motion.div variants={itemVariants}>\n$1\n</motion.div>');
    
    if (modified) {
        fs.writeFileSync(filePath, content, 'utf8');
        console.log('Updated:', filePath);
    }
}

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walkDir(fullPath);
        } else if (file === 'Index.tsx') {
            processFile(fullPath);
        }
    }
}

walkDir('/Users/macm4/Documents/KKN/kknuinsaizu/resources/js/Pages/Admin');

File: add_animations.py
import os
import re

log_lines = []

def process_file(filepath):
    with open(filepath, 'r') as f:
        content = f.read()

    if 'containerVariants' in content:
        return

    modified = False

    # 1. Add import
    if 'import { motion }' not in content:
        content = re.sub(r"(import .*?;)", r"\1\nimport { motion } from 'framer-motion';", content, count=1)
        modified = True

    # 2. Add variants
    variants = """
  const containerVariants = {
    hidden: { opacity: 0 },
    show: {
      opacity: 1,
      transition: { staggerChildren: 0.1, delayChildren: 0.1 },
    },
  };

  const itemVariants = {
    hidden: { opacity: 0, y: 20 },
    show: {
      opacity: 1,
      y: 0,
      transition: { type: 'spring', stiffness: 100, damping: 20 },
    },
  };

"""
    if re.search(r"(return\s*\(\s*<AppLayout[\s\S]*?>)", content):
        content = re.sub(r"(return\s*\(\s*<AppLayout[\s\S]*?>)", variants + r"\1", content, count=1)
        modified = True

    # 3. Replace main container right after <AppLayout ...>
    match = re.search(r'(<AppLayout[^>]*>[\s\S]*?)(<div\s+className="[^"]+">)([\s\S]*)', content)
    if match:
        pre = match.group(1)
        div = match.group(2)
        rest = match.group(3)
        div = div.replace('<div', '<motion.div variants={containerVariants} initial="hidden" animate="show"')
        
        # Now find the last </div> in `rest` before `</AppLayout>`
        rest_split = rest.rsplit('</div>', 1)
        if len(rest_split) == 2:
            rest = rest_split[0] + '</motion.div>' + rest_split[1]
            content = pre + div + rest
            modified = True

    # 4. Wrap inner elements: <PageHeader, <ContentPanel
    if modified:
        content = re.sub(r"(<PageHeader[\s\S]*?<\/PageHeader>)", r"<motion.div variants={itemVariants}>\n\1\n</motion.div>", content)
        content = re.sub(r"(<ContentPanel[\s\S]*?<\/ContentPanel>)", r"<motion.div variants={itemVariants}>\n\1\n</motion.div>", content)
        # for stats grids: <div className="grid... gap-4"> ... </div>
        # this one we can just manually replace `<div className="grid...` with `<motion.div variants={itemVariants}...`
        content = re.sub(r'(<div\s+className="grid[^"]*gap-4[^"]*">)', r'<motion.div variants={itemVariants} className="\g<1>">'.replace('className="<div className="', 'className="'), content)
        content = re.sub(r'(<motion\.div variants=\{itemVariants\} className="grid[^"]*gap-4[^"]*">[\s\S]*?)(</div>)', r'\1</motion.div>', content)

        with open(filepath, 'w') as f:
            f.write(content)
        log_lines.append(f"Updated: {filepath}")

for root, dirs, files in os.walk('/Users/macm4/Documents/KKN/kknuinsaizu/resources/js/Pages/Admin'):
    for file in files:
        if file == 'Index.tsx':
            process_file(os.path.join(root, file))

with open('/Users/macm4/Documents/KKN/kknuinsaizu/add_animations.log', 'w') as f:
    f.write('\n'.join(log_lines))
